crates/pathfinder/src/state/sync/l2.rs

```diff
@@ -265,10 +265,10 @@ async fn download_block(
                 Ok((block, verify_result))
             });
             let (block, verify_result) = verify_hash.await.context("Verify block hash")??;
-            if verify_result == VerifyResult::Mismatch {
-                let block_number = block.block_number;
-                tracing::warn!(?block_number, block_hash = ?expected_block_hash, "Block hash mismatch");
-            }
+            anyhow::ensure!(
+                verify_result != VerifyResult::Mismatch,
+                "Block hash mismatch"
+            );
```

Comment:
Heh. I already did this on a branch. The issue is this effectively breaks all tests which use bogus blocks when setting things up, so this will have to be conditional _not_ to run in tests.

Reply:
Yeah I recall seeing it :) I'm busy manually fixing the fake block hashes so that the tests can pass.